src/Features/LanguageServer/Protocol/Handler/Definitions/AbstractGoToDefinitionHandler.cs

```diff
@@ -62,8 +63,17 @@ protected async Task<LSP.Location[]> GetDefinitionAsync(LSP.TextDocumentPosition
                 {
                     if (!typeOnly || symbol is ITypeSymbol)
                     {
+                        // In order for VisualStudioLspSolutionProvider.GetDocuments to successfully return documents from the metadata workspace, we
+                        // need to open the metadata file on the host and add it to the metadata workspace.
+                        // The file is silently opened on the host, meaning that the host user may not be even aware that the file has been opened.
+                        // This matches the current behavior when opening files on the client, in which opening a file does not also visibly open it
+                        // on the server.
```

Comment:
Why not just add the project to the MetadataAsSourceWorkspace and just not open it in a buffer? Then we don't have to jump to the UI thread in the first place, right? Put another way, is the open-a-file-invisibly thing being done _just_ to make this work, or is it also required for other LSP/Live Share reasons?

Reply:
> is the open-a-file-invisibly thing being done just to make this work

Pretty much. :(

> Why not just add the project to the MetadataAsSourceWorkspace and just not open it in a buffer?

I've tried adding the project to the MetadataAsSourceWorkspace without opening anything, but for some reason, the metadata workspace's `CurrentSolution` won't contain the added project unless we explicitly open one of the project's documents. Could this possibly be a bug?

Reply:
@allisonchou: it looks like today we only are adding the project to the workspace if the file is open, but that could be changed. There's no fundamental reason why it's done the way it is today.